File: apps/ios/PassAlarm/Domain/UseCase/UpdateAppSettingsUseCase.swift
import Foundation

final class UpdateAppSettingsUseCase: Sendable {
    private let appSettingsRepository: AppSettingsRepositoryProtocol
    private let reschedule: RescheduleNextNUseCase

    init(appSettingsRepository: AppSettingsRepositoryProtocol,
         reschedule: RescheduleNextNUseCase) {
        self.appSettingsRepository = appSettingsRepository
        self.reschedule = reschedule
    }

    func execute(_ settings: AppSettings) async throws {
        await appSettingsRepository.save(settings)
        try await reschedule.execute()
    }
}

File: apps/android/app/src/main/kotlin/com/nogeass/passalarm/domain/usecase/UpdateAppSettingsUseCase.kt
package com.nogeass.passalarm.domain.usecase

import com.nogeass.passalarm.domain.model.AppSettings
import com.nogeass.passalarm.domain.repository.AppSettingsRepository
import javax.inject.Inject

class UpdateAppSettingsUseCase @Inject constructor(
    private val appSettingsRepository: AppSettingsRepository,
    private val reschedule: RescheduleNextNUseCase
) {
    suspend fun execute(settings: AppSettings) {
        appSettingsRepository.save(settings)
        reschedule()
    }
}
